cmake-tool/helpers/rootserver.cmake

```diff
@@ -101,37 +97,33 @@ function(DeclareRootserver rootservername)
             else()
                 set(march rv64imafdc)
             endif()
-            if(UseRiscVBBL)
-                # Package up our final elf image into the Berkeley boot loader.
+            if(UseRiscVOpenSBI)
+                # Package up our final elf image into OpenSBI.
                 # The host string is extracted from the cross compiler setting
                 # minus the trailing '-'
                 if("${CROSS_COMPILER_PREFIX}" STREQUAL "")
                     message(FATAL_ERROR "CROSS_COMPILER_PREFIX not set.")
                 endif()
 
-                string(
-                    REGEX
-                    REPLACE
-                        "^(.*)-$"
-                        "\\1"
-                        host
-                        "${CROSS_COMPILER_PREFIX}"
-                )
-                get_filename_component(host ${host} NAME)
+                if("${KernelOpenSBIPlatform}" STREQUAL "")
+                    message(FATAL_ERROR "KernelOpenSBIPlatform not set.")
+                endif()
+
                 file(GLOB_RECURSE deps)
                 add_custom_command(
-                    OUTPUT "${CMAKE_BINARY_DIR}/bbl/bbl"
-                    COMMAND mkdir -p ${CMAKE_BINARY_DIR}/bbl
+                    OUTPUT
+                        "${OPENSBI_PATH}/build/platform/${KernelOpenSBIPlatform}/firmware/fw_payload.elf"
```

Comment:
I unhappy with having twice "${OPENSBI_PATH}/build/platform/${KernelOpenSBIPlatform}/firmware/fw_payload.elf" in the CMake code. This should be an internal helper variable, as this make maintenance easier.

Can we do it this way:
```
set(
    OPENSBI_FW_PAYLOAD_ELF 
    "${OPENSBI_PATH}/build/platform/${KernelOpenSBIPlatform}/firmware/fw_payload.elf"
)
add_custom_command(
   OUTPUT "${OPENSBI_FW_PAYLOAD_ELF}"
   ....
   DEPENDS "${elf_target_file}"
)
# overwrite elf_target_file, it's the the OpenSBI payload ELF file now
set(elf_target_file "${OPENSBI_FW_PAYLOAD_ELF}")
```